components/patient-data-sharing/share-protocol/src/main/java/org/phenotips/data/shareprotocol/ShareProtocol.java

```diff
@@ -165,6 +165,10 @@
 
     public static final String SERVER_JSON_KEY_NAME_ERROR_ACTIONFAILED = "action_failed";
 
+    public static final String SERVER_JSON_KEY_NAME_USERNAME_NOT_IN_CANONICAL_FORM = "username_not_canonical";
+
+    public static final String SERVER_JSON_KEY_NAME_USERNAME_SUGGESTED_CANONICAL_FORM = "username_suggested_canonical";
```

Comment:
I'm not a fan of these field names. In 1 year we'll be scratching our heads wondering what this is all about. I understand that you're incorporating the terminology used by XWiki, but TBH that's not the best either.

I would name these fields in plain English `provided_username` (or `entered_username`) and `expected_username` (or `deduced_username`), similar to your variable names in `DefaultReceivePatientData.java` lines 230-231.

Reply:
The first key is used to give more details about a login problem, so what about naming it `username_not_accepted`? Possibly with a `true\false` instead of the entered username? (but can be either). And naming the other one `expected_username` like you suggested